hirundo/dataset_optimization.py

```diff
@@ -229,51 +416,71 @@ def delete_by_id(dataset_id: int) -> None:
+        This can either be set manually or by creating the `StorageConfig` instance via the `OptimizationDataset`'s
         `create` method
         """
-        if storage_integration:
-            if not self.storage_integration_id:
-                raise ValueError("No storage integration has been created")
-            StorageIntegration.delete_by_id(self.storage_integration_id)
-        if not self.dataset_id:
+        if storage_config:
+            if not self.storage_config_id:
+                raise ValueError("No storage config has been created")
+            StorageConfig.delete_by_id(self.storage_config_id)
+        if not self.id:
             raise ValueError("No dataset has been created")
-        self.delete_by_id(self.dataset_id)
+        self.delete_by_id(self.id)
 
-    def create(self) -> int:
+    def create(
+        self,
+        organization_id: typing.Optional[int] = None,
+        replace_if_exists: bool = False,
+    ) -> int:
         """
         Create a `OptimizationDataset` instance on the server.
-        If `storage_integration_id` is not set, it will be created.
+        If the `storage_config_id` field is not set, the storage config will also be created and the field will be set.
+
+        Args:
+            organization_id: The ID of the organization to create the dataset for.
+            replace_if_exists: If True, the dataset will be replaced if it already exists
+                (this is determined by a dataset of the same name in the same organization).
+
+        Returns:
+            The ID of the created `OptimizationDataset` instance
         """
-        if not self.dataset_storage:
+        if self.storage_config is None and self.storage_config_id is None:
             raise ValueError("No dataset storage has been provided")
-        if (
-            self.dataset_storage
-            and self.dataset_storage.storage_integration
-            and not self.storage_integration_id
+        elif self.storage_config and self.storage_config_id is None:
+            if isinstance(self.storage_config, ResponseStorageConfig):
+                self.storage_config_id = self.storage_config.id
+            elif isinstance(self.storage_config, StorageConfig):
+                self.storage_config_id = self.storage_config.create(
+                    replace_if_exists=replace_if_exists,
+                )
+        elif (
+            self.storage_config is not None
+            and self.storage_config_id is not None
+            and (
+                not isinstance(self.storage_config, ResponseStorageConfig)
+                or self.storage_config.id != self.storage_config_id
+            )
         ):
-            self.storage_integration_id = (
-                self.dataset_storage.storage_integration.create()
+            raise ValueError(
+                "Both `storage_config` and `storage_config_id` have been provided. Storage config IDs do not match."
```

Comment:
We don't check because there is nothing to do. That is, if `self.storage_config_id` is set, then we can continue and create the dataset without any issues. Does that make sense @mishana ? Do you want some change here?

Reply:
Got it